Building `systest` fails with nightly `rustc`

`systest` uses `ctest`, which uses `syntex_syntax2`, which uses `extprim`:

```
    Compiling syntex_pos v0.59.1
error[E0557]: feature has been removed
  --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/extprim-1.7.1/src/lib.rs:58:81
   |
58 | #![cfg_attr(extprim_channel="unstable", feature(llvm_asm, test, specialization, const_fn))]
   |                                                                                 ^^^^^^^^ feature has been removed
   |
   = note: split into finer-grained feature gates
```

A fix PR is pending upstream (https://github.com/kennytm/extprim/pull/21).

--- a/systest/Cargo.toml
+++ b/systest/Cargo.toml
@@ -13,4 +13,4 @@ cubeb-sys = { path = "../cubeb-sys" }
 libc = "0.2"
 
 [build-dependencies]
-ctest = "0.2"
+ctest2 = "0.4"

--- a/systest/build.rs
+++ b/systest/build.rs
@@ -1,4 +1,4 @@
-extern crate ctest;
+extern crate ctest2;
 
 use std::env;
 use std::path::PathBuf;
@@ -6,7 +6,7 @@ use std::path::PathBuf;
 fn main() {
     let root = PathBuf::from(env::var_os("DEP_CUBEB_ROOT").unwrap());
 
-    let mut cfg = ctest::TestGenerator::new();
+    let mut cfg = ctest2::TestGenerator::new();
 
     // Include the header files where the C APIs are defined
     cfg.header("cubeb.h")

